### DISCUSSION TITLE
using the output of i.display_kb() execute qtile commands

### DISCUSSION THREAD

**Discussion by alezunino** [qtile/qtile#4456] [Q&A] [answered]
Hi,
I use InteractiveCommandClient().display_kb() to provide a simple help using rofi. To properly complete the help I would like to execute the command when selecting an entry. This is the output of InteractiveCommandClient().display_kb():
`KeySym                  Mod             Command

Return                  mod4, shift     spawn('xdg-open $HOME', shell = True)

Tab                     mod1            next_group(skip_empty = True)

w                       mod4            to_screen(0)`

The idea is to select "Command". How can I send these Commands to qtile for executing them? Is it possible?
Thanks

**Comment by ervinpopescu** [comment, marked as answer]
I don't think this can be reliably done. The way I deal with keybindings is I save to a `json` file using `jsonpickle` every time qtile is run, reloaded or restarted (see [this](https://github.com/ervinpopescu/dots/blob/lenovo/configs/.config/qtile/modules/groups.py#L100-L102)) and then read the file whenever I need to (for example [here](https://github.com/ervinpopescu/dots/blob/lenovo/configs/bin/qtilekeys.py#L21-L22)). Take a look at how I parse the `keys` and let me know if this is close to what you want.

**Reply by alezunino** [reply]
awesome! thanks!
